modules/infra/control/netlink.c

```diff
@@ -200,6 +200,59 @@ int netlink_del_route(const char *ifname, uint32_t table) {
 	return netlink_add_del_route(ifname, table, false);
 }
 
+static int netlink_add_del_addr(const char *ifname, const void *addr, size_t addr_len, bool add) {
+	char buf[NLMSG_SPACE(
+		sizeof(struct ifaddrmsg) + 2 * NLA_SPACE(sizeof(struct rte_ipv6_addr))
+		+ // IFA_LOCAL + IFA_ADDRESS
+		NLA_SPACE(IFNAMSIZ) // IFA_LABEL
+	)];
+	bool is_ipv4 = (addr_len == sizeof(ip4_addr_t));
+	struct nlmsghdr *nlh;
```

Comment:
Maybe the address add functions could be merged in the `infra: add netlink helpers using libmnl` commit.